src/consensus.rs

```diff
@@ -80,6 +80,19 @@ impl<T: Proposition> Consensus<T> {
         self.sign_vote(vote)
     }
 
+    pub fn contains_new_vote_from_voter(&self, signed_vote: &SignedVote<T>) -> bool {
+        if let Some(previous_vote_from_voter) = self.votes.get(&signed_vote.voter) {
+            if signed_vote == previous_vote_from_voter {
+                false
+            } else {
+                signed_vote.supersedes(previous_vote_from_voter)
+            }
```

Comment:
```suggestion
            signed_vote != previous_vote_from_voter
              && signed_vote.supersedes(previous_vote_from_voter)
```

We can move this into a `SignedVote::strict_supersedes(..)` method, i.e. the equiv. of `>` instead of `>=`